libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1854,6 +1886,15 @@ std::optional<SyncBlock> PbftManager::processSyncBlock() {
     return std::nullopt;
   }
 
+  // Check reward votes
+  vote_mgr_->updateRewardVotes(sync_block.first.pbft_blk->getPeriod() - 1);
```

Comment:
I also dont get why you call updateRewardVotes here, why dont you call it after we push/process new pbft block ? You would not need to get previous period_data at all this way

Reply:
@gaoqi8649

Reply:
Please see https://github.com/Taraxa-project/taraxa-node/pull/1670#issuecomment-1113847745

Reply:
In the current reward votes queue design, the reward votes queue only include additional cert votes those are not include in previous period PBFT block. Add the reward votes to previous period cert votes in DB here, because that will use previous period cert votes to compare reward votes hashes later.

Do you think reward votes queue should include all reward votes objects, not only additional one that not include in previous period?